Makefile

```diff
@@ -335,26 +335,68 @@ shadow-server: export TARGET := clojure
 shadow-server:##@ Start shadow-cljs in server mode for watching
 	yarn shadow-cljs server
 
+test: export TARGET := clojure
+test: export SHADOW_OUTPUT_TO := target/test/test.js
+test: export SHADOW_NS_REGEXP := .*-test$$
+test: ##@test Run all Clojure tests
+	yarn install
+	yarn shadow-cljs compile mocks && \
+	yarn shadow-cljs compile test && \
+	node --require ./test-resources/override.js target/test/test.js
```

Comment:
Code duplication with `unit-test` and `integration-test`.

I would move to a shared task or script.

Reply:
@yakimant, I agree, and I was anticipating this kind of comment in this PR. Since you rejected the PR I will refactor. Thanks for the feedback 👍🏼 

[Just my opinion] I personally don't like makefiles to create interfaces for development, it's always the same story, they end up being a layer of indirection to scripts. Now, using make to help efficiently build source code is a different story, but that's not what our targets do. I worked in a few projects that used makefiles only for that purpose and the rest were all scripts following some common conventions. It was pretty nice! Eventually one project evolved to have an internal CLI, but okay, that can be overkill for us now.

Reply:
Why don't you like make for the purpose of interface for usefull scripts?
I find it useful - you just type `make` and you see all the options.

What would be the alternative?

You can always place your code to the scripts and call them directly if you don't want to use make.

Reply:
Thanks for asking @yakimant. First of all, I don't feel strong about replacing `make`, but I felt like sharing anyway. Unconsciously I probably just wanted to hear your perspective :)

> What would be the alternative?

The alternative could be `just` (https://github.com/casey/just), which is described as a *command runner* (which is what we use `make` for in status-mobile). I've been trying to use it in personal projects to get the feeling of the good and bad things. So far, it's quite smooth coming from make.

> I find it useful - you just type make and you see all the options.

Do you mean completion at point? Yes, that's very useful. Interestingly, our Makefile contains cryptic code to support `make help`, that comes OTB with `just`.

Another alternative for a group of Clojure devs is to replace scripts and/or some/all make targets with babashka behind a CLI (for discoverability). This wouldn't fly well for others that don't know Clojure. It could be in another language. As I said before, it sounds overkill, but it could promote a better DX over time (which was the case in my past experience)

In my systems I don't even have `make` available by default, although it's an essential piece of software, nowadays I also install `just`.

> You can always place your code to the scripts and call them directly if you don't want to use make

Yes, but then it's like I said, from my personal perspective, `make` becomes an unnecessary piece of software when it's used mostly as an indirection layer to scripts.

If we only used scripts and a basic CLI for discoverability (optional) it would work just as well, or at lest that's my theory :)